src/Stratis.Bitcoin.IntegrationTests/Wallet/SendingTransactionOverPolicyByteLimitSpecification_Steps.cs

```diff
@@ -0,0 +1,119 @@
+        }
+
+        private void two_connected_nodes()
+        {
+            this.nodes = this.nodeGroupBuilder
+                .StratisPowNode(NodeOne).Start().NotInIBD().WithWallet(WalletName, WalletPassword)
+                .StratisPowNode(NodeTwo).Start().NotInIBD().WithWallet(WalletName, WalletPassword)
+                .WithConnections()
+                    .Connect(NodeOne, NodeTwo)
+                    .AndNoMoreConnections()
+                .Build();
+        }
+
+        private void node1_builds_undersize_tx_to_send_to_node2()
+        {
+            Node1BuildsTransactionToSendToNode2(UnderPolicyByteLimitUnspentTransactionOutputs);
+        }
+
+        private void node1_builds_oversize_tx_to_send_to_node2()
+        {
+            Node1BuildsTransactionToSendToNode2(OverPolicyByteLimitUnspentTransactionOutputs);
+        }
+
+        private void Node1BuildsTransactionToSendToNode2(int txoutputs)
+        { 
+            this.CoinBaseMaturity = (int) this.nodes[NodeOne].FullNode.Network.Consensus.Option<PowConsensusOptions>().CoinbaseMaturity;
+
+            this.MineBlocks(this.nodes[NodeOne]);
+
+            var nodeTwoAddresses = this.nodes [NodeTwo].FullNode.WalletManager().GetUnusedAddresses(new WalletAccountReference(WalletName, WalletAccountName), txoutputs);
+
+            var nodeTwoRecipients = nodeTwoAddresses.Select(address => new Recipient
+            {
+                ScriptPubKey = address.ScriptPubKey,
+                Amount = Money.COIN
+                }).ToList();
+
+            this.transactionBuildContext = SharedSteps.CreateTransactionBuildContext(WalletName, WalletAccountName, WalletPassword, nodeTwoRecipients, FeeType.Medium, 101);
+
+            try
+            {
+                this.transaction = this.nodes[NodeOne].FullNode.WalletTransactionHandler().BuildTransaction(this.transactionBuildContext);
+                this.transactionFee = this.nodes[NodeOne].GetFee(this.transactionBuildContext);
+                this.nodes[NodeOne].FullNode.NodeService<WalletController>().SendTransaction(new SendTransactionRequest(this.transaction.ToHex()));
+            }
+            catch (Exception e)
+            {
+                this.caughtException = e;
+            }
+        }
+
+        private void transaction_is_just_under_100KB_limit()
+        {
+            this.transaction.GetSerializedSize().Should().BeInRange(99000, 100000);
+        }
+
+        private void node1_fails_with_oversize_tx_wallet_error()
+        {
+            this.caughtException.Should().BeOfType<WalletException>().Which.Message.Should().Contain("Transaction's size is too high");
+        }
+
+        private void node1_succeeds_sending_tx_to_node2()
+        {
+            this.caughtException.Should().BeNull();
+        }
+
+        private void MineBlocks(CoreNode node)
+        {
+            this.sharedSteps.MineBlocks(this.CoinBaseMaturity * 2, node, WalletAccountName, WalletName, WalletPassword);
```

Comment:
not sure you need to mine coinbase maturity * 2. Probably just maturity + 1 so that the coins from the first block can be spent

Reply:
*2 means you are mining 300 blocks so that the first 150 blocks are spendable, are you sure you need 7500 bitcoins for the test?